thetis/timeintegrator.py

```diff
@@ -8,407 +8,50 @@
-        # stage 2
-        self.solution_old.assign(solution + 0.25*self.K0 + 0.25*self.K1)
-        for k in self.fields:  # set args to t+dt/2
-            if k in self.funcs_nplushalf:
-                self.fields[k].assign(self.funcs_nplushalf[k])
-            else:
-                self.fields[k].assign(0.5*self.fields_new[k] + 0.5*self.fields_old[k])
-        if update_forcings is not None:
-            update_forcings(t+dt/2)
-        self.solver_k2.solve()
-        # final solution
-        solution.assign(solution + (1.0/6.0)*self.K0 + (1.0/6.0)*self.K1 +
-                        (2.0/3.0)*self.K2)
-
-        # store old values
-        for k in self.fields_old:
-            self.fields_old[k].assign(self.fields_new[k])
-        self.solution_old.assign(solution)
-
-    def solve_stage(self, i_stage, t, dt, solution, update_forcings=None):
-        if i_stage == 0:
-            # stage 0
-            self.solution_n.assign(solution)
-            self.solution_old.assign(solution)
-            if update_forcings is not None:
-                update_forcings(t)
-            self.solver_k0.solve()
-            solution.assign(self.solution_n + self.K0)
-        elif i_stage == 1:
-            # stage 1
-            self.solution_old.assign(solution)
-            if update_forcings is not None:
-                update_forcings(t+dt)
-            self.solver_k1.solve()
-            solution.assign(self.solution_n + 0.25*self.K0 + 0.25*self.K1)
-        elif i_stage == 2:
-            # stage 2
-            self.solution_old.assign(solution)
-            for k in self.fields:  # set args to t+dt/2
-                if k in self.funcs_nplushalf:
-                    self.fields[k].assign(self.funcs_nplushalf[k])
-                else:
-                    self.fields[k].assign(0.5*self.fields_new[k] + 0.5*self.fields_old[k])
-            if update_forcings is not None:
-                update_forcings(t+dt/2)
-            self.solver_k2.solve()
-            # final solution
-            solution.assign(self.solution_n + (1.0/6.0)*self.K0 +
-                            (1.0/6.0)*self.K1 + (2.0/3.0)*self.K2)
-
+        pass
 
-class SSPRK33Stage(TimeIntegrator):
-    """
-    3rd order Strong Stability Preserving Runge-Kutta scheme, SSP(3,3).
-    This class only advances one step at a time.
+    @abstractproperty
+    def initialize(self, init_solution):
+        """Initializes the time integrator"""
+        pass
 
-    This scheme has Butcher tableau
-    0   |
-    1   | 1
-    1/2 | 1/4 1/4
-    ---------------
-        | 1/6 1/6 2/3
 
-    CFL coefficient is 1.0
-    """
-    def __init__(self, equation, solution, fields, dt, bnd_conditions=None, solver_parameters={}):
-        """Creates forms for the time integrator"""
-        super(SSPRK33Stage, self).__init__(equation, solver_parameters)
-        self.explicit = True
-        self.CFL_coeff = 1.0
-        self.n_stages = 3
+class TimeIntegrator(TimeIntegratorBase):
```

Comment:
It seems like TimeIntegratorBase is an unnecessary class?  You could just make TimeIntegrator the meta class and add the `advance` and `initialise` abstract methods to it?  In passing, given that they are methods, not properties, you probably want to decorate them with `@abstractmethod`, rather than `@abstractproperty` (also importable from `abc`).